app/src/renderer/components/network/PageBlock.vue

```diff
@@ -81,21 +81,37 @@ export default {
       ]
     },
     blockTitle({ num, block } = this) {
+      if (!block.block) return `n/a`
       return `#` + num.prettyInt(block.block.header.height)
     },
     blockTime({ moment, block } = this) {
+      if (!block.block) return `n/a`
       return moment(block.block.header.time).format(`MMM Do YYYY, HH:mm:ss`)
     }
   },
   watch: {
-    $route: `getBlock`
+    "$route.params.height": function() {
+      this.getBlock()
+    }
   },
   mounted() {
     this.getBlock()
   },
   methods: {
-    async getBlock() {
-      await this.$store.dispatch(`queryBlockInfo`, this.$route.params.height)
+    async getBlock({ $store, $route, $router, lastHeader } = this) {
+      // query first for the block so we don't fail if the user started from this route and hasn't received any lastHeader yet
+      const blockInfo = await $store.dispatch(
+        `queryBlockInfo`,
+        $route.params.height
+      )
+
+      if (
+        !blockInfo &&
+        Number($route.params.height) > Number(lastHeader.height)
```

Comment:
shouldn't we have this `if` condition before the dispatch ?

Reply:
I even commented the reason above ;)
> query first for the block so we don't fail if the user started from this route and hasn't received any lastHeader yet